test-specifications/token-service/tokenAssociateTransaction.md

```diff
@@ -0,0 +1,142 @@
+
+#### JSON Request Example
+
+```json
+{
+  "jsonrpc": "2.0",
+  "id": 99232,
+  "method": "associateToken",
+  "params": {
+    "accountId": "0.0.2533",
+    "tokenIds": [
+      "0.0.579680",
+      "0.0.90649"
+    ],
+    "commonTransactionParams": {
+      "signers": [
+        "302e020100300506032b65700422042031f8eb3e77a04ebe599c51570976053009e619414f26bdd39676a5d3b2782a1d"
+      ]
+    }
+  }
+}
+```
+
+#### JSON Response Example
+
+```json
+{
+  "jsonrpc": "2.0",
+  "id": 99232,
+  "result": {
+    "status": "SUCCESS"
+  }
+}
+```
+
+### **Token IDs:**
+
+- The IDs of the tokens to associate.
+
+| Test no | Name                                                  | Input                                                                                                                                              | Expected response                                                                                   | Implemented (Y/N) |
+|---------|-------------------------------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------|-----------------------------------------------------------------------------------------------------|-------------------|
+| 1       | Associates no tokens with an account                  | accountId=<CREATED_ACCOUNT_ID>, commonTransactionParams.signers=[<CREATED_ACCOUNT_PRIVATE_KEY>]                                                    | The token association succeeds and no associations are made.                                        | N                 |
+| 2       | Associates a token that doesn't exist with an account | accountId=<CREATED_ACCOUNT_ID>, tokenIds=["123.456.789"], commonTransactionParams.signers=[<CREATED_ACCOUNT_PRIVATE_KEY>]                          | The token association fails with an INVALID_TOKEN_ID response code from the network.                | N                 |
+| 3       | Associates a token that is deleted with an account    | accountId=<CREATED_ACCOUNT_ID>, tokenIds=[<DELETED_TOKEN_ID>], commonTransactionParams.signers=[<CREATED_ACCOUNT_PRIVATE_KEY>]                     | The token association fails with an TOKEN_WAS_DELETED response code from the network.               | N                 |
+| 4       | Associates a token that is empty with an account      | accountId=<CREATED_ACCOUNT_ID>, tokenIds=[""], commonTransactionParams.signers=[<CREATED_ACCOUNT_PRIVATE_KEY>]                                     | The token association fails with an SDK internal error.                                             | N                 |
+| 5       | Associates a token twice with an account              | accountId=<CREATED_ACCOUNT_ID>, tokenIds=[<CREATED_TOKEN_ID>, <CREATED_TOKEN_ID>], commonTransactionParams.signers=[<CREATED_ACCOUNT_PRIVATE_KEY>] | The token association fails with an TOKEN_ID_REPEATED_IN_TOKEN_LIST response code from the network. | N                 |
```

Comment:
Would it be reasonable to add cases here for multiple tokens in a single transaction?
It is supposed to be possible to associate several tokens at a time, and it might be worth testing scenarios including, for example
3 valid tokens
2 valid tokens and one invalid token
3 tokens, two of which are the same ID
3 tokens, one of which is deleted.

I don't actually know the results for those scenarios (e.g. will the system associate the valid tokens, even if the list contains an invalid token, or will the entire transaction be rejected?); which suggests that testing them might be valuable.

Reply:
Added the additional outlined test cases